src/field_theory/complete_ordered.lean

```diff
@@ -0,0 +1,594 @@
+@[symm] protected def symm {R S : Type*}
+[has_mul R] [has_add R] (r : R → R → Prop)
+[has_mul S] [has_add S] (s : S → S → Prop)
+  (f : r ≃+*o s) : s ≃+*o r :=
+{ ..f.to_ring_equiv.symm,
+  ..f.to_rel_iso.symm, }
+
+/-- Composition of two ordered ring isomorphisms is an order isomorphism. -/
+@[trans] protected def trans {R S T : Type*} [has_mul R] [has_add R] (r : R → R → Prop)
+  [has_mul S] [has_add S] (s : S → S → Prop)
+  [has_mul T] [has_add T] (t : T → T → Prop)
+    (f₁ : r ≃+*o s) (f₂ : s ≃+*o t) : r ≃+*o t :=
+{ ..f₁.to_ring_equiv.trans f₂.to_ring_equiv,
+  ..f₁.to_rel_iso.trans f₂.to_rel_iso, }
+
+end ordered_ring_equiv
+
+/-- A field which is both linearly ordered and conditionally complete with respect to the order,
+    this provides a model of the reals. -/
+class conditionally_complete_linear_ordered_field (F : Type*)
+  extends linear_ordered_field F, conditionally_complete_linear_order F
+
+-- TODO conditionally_complete_lattice or conditioanlly_complete_linear order?
+
+/-- The reals are a conditionally complete linearly ordered field. -/
+instance : conditionally_complete_linear_ordered_field ℝ := {
+  ..real.linear_ordered_field,
+  ..real.conditionally_complete_linear_order }
+
+lemma exists_rat_sqr_btwn_rat_aux (x y : ℝ) (h : x < y) (hx : 0 ≤ x) :
+  ∃ q : ℚ, 0 ≤ q ∧ x < q^2 ∧ ↑q^2 < y :=
+begin
+  have : x.sqrt < y.sqrt :=
+  begin
+    rw real.sqrt_eq_rpow,
+    apply real.rpow_lt_rpow hx h one_half_pos,
+  end,
+  obtain ⟨q, hqx, hqy⟩ := exists_rat_btwn this,
+  have hy : (0 : ℝ) ≤ y := by linarith,
+  have hq : (0 : ℝ) ≤ q :=
+  begin
+    transitivity x.sqrt,
+    exact real.sqrt_nonneg x,
+    exact le_of_lt hqx,
+  end,
+  have hq2 := pow_nonneg hq 2,
+  refine ⟨q, _, _, _⟩,
+  { assumption_mod_cast, },
+  { rw ← real.sqrt_lt hx hq2, rw real.sqrt_sqr hq, exact hqx },
+  { rw ← real.sqrt_lt hq2 hy, rw real.sqrt_sqr hq, exact hqy },
+end
+
+theorem exists_rat_sqr_btwn_rat {x y : ℚ} (h : x < y) (hx : 0 ≤ x) :
+  ∃ q : ℚ, 0 ≤ q ∧ x < q^2 ∧ q^2 < y :=
+begin
+  have := exists_rat_sqr_btwn_rat_aux x y,
+  norm_cast at this,
+  apply this; assumption,
+end
+
+/-- There is a rational square between any two elements of an archimedean ordered field -/
+theorem exists_rat_sqr_btwn {F : Type*} [linear_ordered_field F] [archimedean F] {x y : F}
+(h : x < y) (hx : 0 ≤ x) : ∃ q : ℚ, 0 ≤ q ∧ x < q^2 ∧ (q^2 : F) < y :=
+begin
+  obtain ⟨q1, hq1x, hq1y⟩ := exists_rat_btwn h,
+  obtain ⟨q2, hq2x, hq1q2⟩ := exists_rat_btwn hq1x,
+  norm_cast at hq1q2,
+  have : (0 : F) ≤ q2 :=
+  begin
+    transitivity x,
+    exact hx,
+    exact (le_of_lt hq2x),
+  end,
+  obtain ⟨q, hqpos, hq⟩ := exists_rat_sqr_btwn_rat hq1q2 (by exact_mod_cast this),
+  refine ⟨q, hqpos, _, _⟩,
+  { transitivity (q2 : F),
+    exact hq2x,
+    exact_mod_cast hq.1, },
+  { transitivity (q1 : F),
+    exact_mod_cast hq.2,
+    exact hq1y, }
+end
+
+-- example {F : Type u_1} -- TODO library_search output has too many args
```

Comment:
Is this still an issue? If so, please file it on github.

Reply:
It is the same as https://github.com/leanprover-community/lean/issues/394